Crate does not compile on stable Rust < 1.36

Hello,

It seems that your librairie does not compile anymore since the 1.6.0, and also, all dependencies does not compile anymore too! (like test-junit)

```
error[E0506]: cannot assign to `self.state` because it is borrowed

   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/typed-arena-1.6.0/src/lib.rs:461:33

    |

451 |                     ref mut inner_iter,

    |                     ------------------ borrow of `self.state` occurs here

...

461 |                                 self.state = IterMutState::ChunkListRest { index, inner_iter };

    |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ assignment to borrowed `self.state` occurs here


error[E0506]: cannot assign to `self.state` because it is borrowed

   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/typed-arena-1.6.0/src/lib.rs:467:33

    |

451 |                     ref mut inner_iter,

    |                     ------------------ borrow of `self.state` occurs here

...

467 |                                 self.state = IterMutState::ChunkListCurrent { iter };

    |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ assignment to borrowed `self.state` occurs here

```
We are using the **nightly-2019-03-06** rust version.

Do you confirm this error ?
Could you check your publication please ?
:/

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,6 +3,7 @@ language: rust
 
 matrix:
   include:
+    - rust: 1.32.0
     - rust: stable
     - rust: beta
     - rust: nightly

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -445,7 +445,7 @@ impl<'a, T> Iterator for IterMut<'a, T> {
     type Item = &'a mut T;
     fn next(&mut self) -> Option<&'a mut T> {
         loop {
-            match self.state {
+            self.state = match self.state {
                 IterMutState::ChunkListRest {
                     mut index,
                     ref mut inner_iter,
@@ -458,20 +458,18 @@ impl<'a, T> Iterator for IterMut<'a, T> {
                                 let inner_iter = self.chunks.rest[index].iter_mut();
                                 // Extend the lifetime of the individual elements to that of the arena.
                                 let inner_iter = unsafe { mem::transmute(inner_iter) };
-                                self.state = IterMutState::ChunkListRest { index, inner_iter };
-                                continue;
+                                IterMutState::ChunkListRest { index, inner_iter }
                             } else {
                                 let iter = self.chunks.current.iter_mut();
                                 // Extend the lifetime of the individual elements to that of the arena.
                                 let iter = unsafe { mem::transmute(iter) };
-                                self.state = IterMutState::ChunkListCurrent { iter };
-                                continue;
+                                IterMutState::ChunkListCurrent { iter }
                             }
                         }
                     }
                 }
                 IterMutState::ChunkListCurrent { ref mut iter } => return iter.next(),
-            }
+            };
         }
     }
 

